### DISCUSSION TITLE
3.1.0

### DISCUSSION THREAD

**Discussion by julien-truffaut** [julien-truffaut/Monocle#1198] [General]
# What's Changed
* Add `refocus` to all optics as an alias for `.andThen(Focus(...))` (#1184) @kenbot
```scala
case class User(name: String, address: Address)
case class Address(streetNumber: Int, street: Option[Street])

val elise = User("Elise", Address(12, Some(Street("high street"))))

val addressLens: Lens[User, Address] = Focus[User](_.address)
val newLens    : Lens[User, Int]     = addressLens.refocus(_.streetNumber)

newLens.replace(50)(elise)

// or using the applied syntax
import monocle.syntax.all._

elise.focus(_.address).refocus(_.streetNumber)
```
* Add `GenIso` to Scala 3 (#1185) @japgolly
* Add `Iso.fields` (#1197) @kenbot
* `index` is now a direct method for all optics for better Cats compat (#1192) @kenbot
* Avoid indirect call to `AsPrismImpl` (#1186) @nicolasstucki
* Fix #1177 where Focus could not operate on type aliases (#1178) @kenbot

## Dependencies
* Update sbt-scalajs, scalajs-compiler, ... to 1.7.0 (#1187) @scala-steward 
* Update refined, refined-scalacheck to 0.9.27 (#1179) @scala-steward

<hr /><em>This discussion was created from the release <a href='https://github.com/optics-dev/Monocle/releases/tag/v3.0.1'>3.1.0</a>.</em>

**Comment by yanns** [comment]
This release does not seem to be on maven central for scala 2.13: https://repo1.maven.org/maven2/dev/optics/monocle-core_2.13/

**Reply by julien-truffaut** [reply]
Thanks for letting me know. I screw up when setting the tag ... This version should be available later today.

**Reply by yanns** [reply]
It's here: https://repo1.maven.org/maven2/dev/optics/monocle-core_2.13/3.1.0/
Thanks a lot!
